resources/agent.rb

```diff
@@ -52,9 +52,14 @@
       else
         version new_resource.version
         action :install
+        notifies :reload, 'systemd_unit[sensu-agent]', :immediately
```

Comment:
I think we want this notifies for both install and upgrade actions. I am also wondering if it should still be conditional to check the platform that its expected to use `systemd`. For example see:
https://github.com/sensu/sensu-go-chef/pull/132/files#diff-89d880e5aa57a7321ba37094ee4041d29aec8fccdaad2d28bb2dd2f0ab9f3ec4R71
```ruby
if platform?('ubuntu') && node['platform_version'].to_f == 14.04
```

In our case I think we might adapt it like so:
```suggestion
        notifies :reload, 'systemd_unit[sensu-agent]', :immediately if use_systemd?
```

wondering if it might be helpful at this point to make a `systemd?` helper library method to make it a bit easier to use. Not sure all the logic we need but maybe something like this:
```ruby
def use_systemd?
  return false if platform?('windows')
  return false if platform?('ubuntu') && node['platform_version'].to_f == 14.04
  # really wish there was a reliable way to check
  return true if node['init_package'] == 'systemd'
  # for safety if we are not sure we assume we are not systemd
  return false
end
```

Reply:
Thanks for the feedback, I've added the same reload notification to the upgrade action as well.  I don't think explicit checking for systemd is necessary though as it is the default behavior for *nix installations, both agent and backend.  As far as cross platform detection I agree it's a shame there isn't better unification among existing Chef resources.  I think the only other option would be to explicitly define a systemd file, but that seems a bit messy and outside the scope of this cookbook I guess.